R/utils.R

```diff
@@ -4,3 +4,24 @@
 #'
 #' @keywords internal
 inf_dev_extendr_used <- function() "Are you using a development version of {.code extendr}?"
+
+
+#' Silence `{cli}` output
+#'
+#' Use for functions that use cli output that should optionally be suppressed.
+#'
+#' @examples
+#' hello_rust <- function(..., quiet = FALSE) {
+#'   with_quiet(quiet)
+#'   cli::cli_alert_info("This should be silenced when {.code quiet = TRUE}")
+#' }
+#'
+#' hello_rust()
+#' hello_rust(quiet = TRUE)
+#' @keywords internal
+with_quiet <- function(quiet) {
+  if (quiet) withr::local_options(
+    list("cli.default_handler" = function(...) {}),
+    .local_envir = parent.frame(n = 1)
+    )
+}
```

Comment:
How about we add `env = rlang::caller_env()`, like it is [commonly done](https://rlang.r-lib.org/reference/eval_tidy.html)? I am a little bit concerned with this `parent.frame()` call.

Reply:
TIL `caller_env()` is equivalent to `parent.frame()` + 1
``` r
rlang::caller_env
#> function (n = 1) 
#> {
#>     parent.frame(n + 1)
#> }
#> <bytecode: 0x1053d8350>
#> <environment: namespace:rlang>
```

<sup>Created on 2023-04-18 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>